validator/client/src/main/java/tech/pegasys/teku/validator/client/ForkProvider.java

```diff
@@ -43,49 +43,63 @@ public ForkProvider(
+    return currentFork;
   }
 
-  public SafeFuture<ForkInfo> loadForkInfo() {
-    return requestForkInfo()
-        .exceptionallyCompose(
-            error -> {
-              LOG.error("Failed to retrieve current fork info. Retrying after delay", error);
-              return asyncRunner.runAfterDelay(
-                  this::getForkInfo, FORK_RETRY_DELAY_SECONDS, TimeUnit.SECONDS);
-            });
-  }
-
-  private SafeFuture<ForkInfo> requestForkInfo() {
-    return genesisDataProvider.getGenesisValidatorsRoot().thenCompose(this::requestFork);
-  }
-
-  public SafeFuture<ForkInfo> requestFork(final Bytes32 genesisValidatorsRoot) {
+  private SafeFuture<ForkInfo> getForkAndPeriodicallyUpdate() {
     return validatorApiChannel
         .getFork()
         .thenCompose(
             maybeFork -> {
               if (maybeFork.isEmpty()) {
                 LOG.trace("Fork info not available, retrying");
-                return asyncRunner.runAfterDelay(
-                    this::requestForkInfo, FORK_RETRY_DELAY_SECONDS, TimeUnit.SECONDS);
+                return scheduleNextRequest(FORK_RETRY_DELAY_SECONDS);
               }
-              final ForkInfo forkInfo =
-                  new ForkInfo(maybeFork.orElseThrow(), genesisValidatorsRoot);
-              currentFork = Optional.of(forkInfo);
-              // Periodically refresh the current fork info.
-              asyncRunner
-                  .runAfterDelay(this::loadForkInfo, FORK_REFRESH_TIME_SECONDS, TimeUnit.SECONDS)
-                  .reportExceptions();
-              return SafeFuture.completedFuture(forkInfo);
+
+              scheduleNextRequest(FORK_REFRESH_TIME_SECONDS).reportExceptions();
+              return getValidatorsRootAndCreateForkInfo(maybeFork.orElseThrow());
+            })
+        .exceptionallyCompose(
+            error -> {
+              LOG.error("Failed to retrieve current fork info. Retrying after delay", error);
+              return scheduleNextRequest(FORK_RETRY_DELAY_SECONDS);
             });
   }
 
+  private SafeFuture<ForkInfo> getValidatorsRootAndCreateForkInfo(Fork fork) {
+    return genesisDataProvider
+        .getGenesisValidatorsRoot()
+        .thenCompose(
+            root -> {
+              final ForkInfo forkInfo = new ForkInfo(fork, root);
+              // anything waiting on the current future needs to be notified
+              currentFork.complete(forkInfo);
+              currentFork = SafeFuture.completedFuture(forkInfo);
+              return currentFork;
+            });
+  }
+
+  private SafeFuture<ForkInfo> scheduleNextRequest(final long seconds) {
+    return asyncRunner.runAfterDelay(this::getForkAndPeriodicallyUpdate, seconds, TimeUnit.SECONDS);
+  }
+
   @Override
   public String toString() {
     return "ForkProvider{" + "currentFork=" + currentFork + '}';
   }
+
+  @Override
+  protected SafeFuture<?> doStart() {
+    currentFork = getForkAndPeriodicallyUpdate();
```

Comment:
This would overwrite the `SafeFuture` assigned in the constructor without actually completing it.  Probably should make the initial value a failed future if we're going to do this ( probably an `IllegalStateException` given it would mean `getForkInfo` is called before `start` is. Would have to be completely sure it never happens though so may be better to just have this be:
```suggestion
    getForkAndPeriodicallyUpdate().propagateTo(currentFork);
```